File: app/src/main/java/com/example/rakeshvasal/myapplication/Dao/UserDao.java
package com.example.rakeshvasal.myapplication.Dao;




import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.rakeshvasal.myapplication.GetterSetter.User;

import java.util.List;

@Dao
public interface UserDao {

    @Query("Select * from users")
    List<User> getAllUsers();

    @Insert
    void addUser(User user);

    @Query("Select * from users where user_id = :id")
    User getUserById(String id);

    @Query("Select * from users where user_name = :name")
    User getUserByName(String name);

    @Query("DELETE from users")
    void DeleteAllUsers();

    @Query("DELETE from users where user_id = :id")
    void DeleteUserById(String id);
}

File: app/src/main/java/com/example/rakeshvasal/myapplication/Dao/UserDao.kt
package com.example.rakeshvasal.myapplication.Dao

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.example.rakeshvasal.myapplication.DbUser

interface UserDao {

    @Query("SELECT  *  FROM users")
    fun getAllUsers() : List<DbUser>

    @Query("SELECT * FROM users WHERE user_id IN (:userIds)")
    fun getUsersFromIds(userIds : IntArray) : List<DbUser>

    @Insert
    fun insertAll(vararg users: DbUser)

    @Delete
    fun delete(user: DbUser)

    @Update
    fun updateTodo(vararg users: DbUser)
}